### PR TITLE
Story 2 task 4

### PR DESCRIPTION
**Pull request by DanAbsolon**
we implemented the projectTest class

### CODE REVIEW

File: src/Tests/Entities/ProjectTest.php [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+class ProjectTest extends TestCase
+{
+    public function testSuccessConstructor_overdueTrue()
+    {
+        //creating a deadline that is 1 year before the current date
+        $deadline = new DateTime();
+        $dateInterval = new DateInterval('P1Y');
+        $dateInterval->invert = 1;
+        $deadline->add($dateInterval);
```
Comment (robin-hodge) [inline src/Tests/Entities/ProjectTest.php:19]: I really like how you have done this!

File: src/Tests/Entities/ProjectTest.php [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        $deadline = null;
+        $testProject = new Project(1, 'name', 1, $deadline);
+        $actualOutput = $testProject->getIsOverdue();
+        $expectedOutput = null;
+        $this->assertEquals($expectedOutput, $actualOutput); 
+    }
+    
+    public function testSuccessToAssociativeArrayFewerProperties()
+    {
+        $deadline = new DateTime('12/21/2012');
```
Comment (tauroid) [inline src/Tests/Entities/ProjectTest.php]: for readability, we were thinking replace with
`$deadline = DateTime::createFromFormat('d/m/Y', '21/12/2012');`

File: src/Tests/Entities/ProjectTest.php [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        $testProject = new Project(1, 'name', 1, $deadline);
+        $expectedOutput = [
+                            'id' => '1', 
+                            'name' => 'name', 
+                            'client_id' => '1', 
+                            'deadline' => $deadline->format('d/m/Y'),
```
Comment (tauroid) [inline src/Tests/Entities/ProjectTest.php]: and just have `'deadline' => '21/12/2012'` as discussed

**Review by tauroid — CHANGES_REQUESTED** [review]
Looks great just that little readability thing

File: src/Tests/Entities/ProjectTest.php (hunk 2 above)
Comment (DanAbsolon) [reply src/Tests/Entities/ProjectTest.php]: Good idea, it's done!

File: src/Tests/Entities/ProjectTest.php (hunk 3 above)
Comment (DanAbsolon) [reply src/Tests/Entities/ProjectTest.php]: done!

**Review by tauroid — APPROVED** [review]
niceeeeeeeeeeeeeee

**Pull request merged**